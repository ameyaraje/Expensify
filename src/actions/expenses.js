import { v1 as uuidv1 } from 'uuid';

export const addExpense = ({ description = '', note = '', amount = 0, createdAt = 0 } = {}) => {
    return {
        type: 'ADD_EXPENSE',
        expense: {
            id: uuidv1(),
            description: description,
            note: note,
            amount: amount,
            createdAt: createdAt
        }
    };
};

export const removeExpense = ({ id } = {}) => {
    return {
        type: 'REMOVE_EXPENSE',
        id: id
    };
};

export const editExpense = (id, updates) => {
    return {
        type: 'EDIT_EXPENSE',
        id: id,
        updates: updates  
    };
};