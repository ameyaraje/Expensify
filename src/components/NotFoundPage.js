import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => {
    return (
        <div>
            This page does not exist. Please check the route you're on.
            Click <Link to="/">here</Link> to go back
        </div>
    );
};

export default NotFoundPage;