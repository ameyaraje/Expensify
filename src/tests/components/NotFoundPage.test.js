import React from 'react';
import { shallow } from 'enzyme';

import expenses from '../fixtures/expenses';
import NotFoundPage from '../../components/NotFoundPage';

test('should render NotFoundPage', () => {
    const wrapper = shallow(<NotFoundPage />);

    expect(wrapper).toMatchSnapshot();
});