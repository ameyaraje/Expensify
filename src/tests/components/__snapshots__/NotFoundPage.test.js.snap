// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotFoundPage 1`] = `
<div>
  This page does not exist. Please check the route you're on. Click 
  <Link
    to="/"
  >
    here
  </Link>
   to go back
</div>
`;
